design/repo_maintenance_job_config.md

```diff
@@ -0,0 +1,144 @@
+# Repository maintenance job configuration design
+
+## Abstract
+Add this design to make the repository maintenance job can read configuration from a dedicate ConfigMap and make the Job's `PodSpec.Affinity` configurable. 
+
+## Background
```

Comment:
This configuration has larger scope that node affinity, i.e., concurrency, resource limit, etc. So let's add some clarification of what the configuration will be used overall --- something like this configuration is used to make the configuration to the Repository maintenance job spec, so as to control the job pod's behavior (schedule, parallelism, etc.)